odoo/fields.py

```diff
@@ -3047,6 +3057,538 @@ def _record_ids_per_res_model(self, records):
+        else:
+            properties_list_values = self._dict_to_list(properties_values, properties_definition)
+
+        properties_list_values = properties_list_values[:]
+        for properties_value in properties_list_values:
+            if properties_value.get('value') is None:
+                property_type = properties_value.get('type')
+                default = properties_value.get('default')
+                if property_type in ('many2one', 'many2many') and default:
+                    # need to add the model for the child value
+                    # 14 => (14, 'res.partner')
+                    properties_value['value'] = [default, properties_value.get('model')]
+                else:
+                    properties_value['value'] = default or False
+
+        return properties_list_values
+
+    def _get_properties_definition(self, record):
+        """Return the properties definition of the given record."""
+        parent = record[self.parent_field]
+        if parent:
+            return parent.sudo()[self.parent_properties_field]
+
+    @classmethod
+    def _add_display_name(cls, values_list, env, value_keys=('value', 'default')):
+        """Add the "name_get" for each many2one / many2many properties.
+
+        :param values_list: List of properties definition and values
+        :param env: environment
+        """
+        for property_definition in values_list:
+            property_type = property_definition.get('type')
+            property_model = property_definition.get('model')
+            if not property_model:
+                continue
+
+            for value_key in value_keys:
+                property_value = property_definition.get(value_key)
+
+                if property_type == 'many2one' and isinstance(property_value, int):
+                    try:
+                        property_value = env[property_model].browse(property_value).name_get()
+                        property_definition[value_key] = property_value[0] if property_value else False
+                    except AccessError:
+                        # protect from access error message, show an empty name
+                        property_definition[value_key] = (property_value, "")
+
+                elif property_type == 'many2many' and is_list_of(property_value, int):
+                    try:
+                        property_value = env[property_model].browse(property_value).name_get()
+                        property_definition[value_key] = property_value or False
+                    except AccessError:
+                        # protect from access error message, show an empty name
+                        property_definition[value_key] = [(res_id, "") for res_id in property_value]
+
+        return values_list
+
+    @classmethod
+    def _remove_display_name(cls, values_list, value_key='value'):
+        """Remove the display name received bu the web client for the relational properties.
```

Comment:
So you are enabling the web client to send crap?

Reply:
who cares anyway if the marketing is happy? Nah...
![2022-08-21_01-57](https://user-images.githubusercontent.com/16926/185776249-fedcf520-98b0-45b3-8ed8-6847db6b5ecb.PNG)